crates/nu-cli/src/completions/command_completions.rs

```diff
@@ -88,68 +103,41 @@ impl CommandCompletion {
         span: Span,
         offset: usize,
         find_externals: bool,
-        match_algorithm: MatchAlgorithm,
+        options: &CompletionOptions,
     ) -> Vec<SemanticSuggestion> {
         let partial = working_set.get_span_contents(span);
+        let mut matcher = NuMatcher::new(String::from_utf8_lossy(partial), options.clone());
 
-        let filter_predicate = |command: &[u8]| match_algorithm.matches_u8(command, partial);
+        let sugg_span = reedline::Span::new(span.start - offset, span.end - offset);
 
-        let mut results = working_set
-            .find_commands_by_predicate(filter_predicate, true)
-            .into_iter()
-            .map(move |x| SemanticSuggestion {
+        let mut matched_internal = HashSet::new();
+        for (name, description, typ) in working_set.find_commands_by_predicate(|_| true, true) {
```

Comment:
Since the predicate accepted by `find_commands_by_predicate` isn't a `FnMut`, we can't pass in `|x| matcher.add_semantic_suggestion(...)`. This means that it's going to create a vector containing all commands first, rather than only creating a vector of the commands that match. I'm unsure if the hit from this will be noticeable.

Here are the two simplest alternatives I can see:
- Make `find_commands_by_predicate` take in a `FnMut`
- Add a method to working sets that returns an iterator over commands

Reply:
I talked to Ian in Discord and he suggested making `find_commands_by_predicate` take a `FnMut`. Should be good now!